pkg/trace/force_sampler.go

```diff
@@ -38,3 +47,12 @@ func forceTracing(ctx context.Context) context.Context {
 	}
 	return ctx
 }
+
+// sampleAt sets the sample rate for a given trace explicitly. The rate given will be
+// parsed as 1/rate to actually determine the sample rate.
+func sampleAt(ctx context.Context, rate uint) context.Context {
+	if t := trace.GetTraceFromContext(ctx); t != nil {
+		t.AddField(fieldSampleTrace, rate)
+	}
+	return ctx
+}
```

Comment:
FWIW, I'd not have done it this way but this is ok.

I'd have done it as a log marshaler:

```golang

type TraceSampleRate  uint
func (t TraceSampleRate) MarshalLog(addField func(k string, v interface{})) {
    addField(fieldSampleTrace, rate)
}
```

I believe if we did this way, you can just do:

```golang
    ctx = trace.StartCall(ctx, "foo", log.TraceSampleRate(1000))
```

But we don't expect to use this a whole lot, so it doesn't matter a whole lot.